### PR TITLE
INFRA-1899: Refactor confidential-identities Pipeline to reduce duplication and add security scanning 

### PR DESCRIPTION
**Pull request by ronanbrowne**
The PR refactors this repo to use our standard `multi-branch` pipeline approach and reduces the need for duplication across multiple Jenkins files in this repo.

These changes will support and streamline upcoming scheduled release activity for the repository,

Changes:
- remove 2nd duplicate Jenkins file in ./ci/release/jenkisnfile
- Use our standard java 8 docker image when building
- Remove redundant docker file in the repo that was used previously
- Add Nexus (this also requires publish to maven local stage)
- add new property `versionSuffix` for specifying RC/HC etc doing release process 
- Various minor tidy ups on existing Jenkins logic
- Add Gradle Enterprise plugin and associated logic to produce Build Scans 

Tag build tested here: https://ci01.dev.r3.com/job/Corda-SDKs/job/confidential-identities/view/tags/job/release-test-tag/

Once merged the new multi branch job [here ](https://ci01.dev.r3.com/job/Corda-SDKs/job/confidential-identities/)will relace the multiple existing jobs in the subfolders [here](https://ci01.dev.r3.com/job/Corda-SDKs/)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,31 +1,129 @@
...
+@Field
+String mavenLocal = 'tmp/mavenlocal'
+
+def nexusDefaultIqStage = "build"
+
+/**
+ * make sure calculated default value of NexusIQ stage is first in the list
+ * thus making it default for the `choice` parameter
+ */
+def nexusIqStageChoices = [nexusDefaultIqStage].plus(
```
Comment (yarivus) [inline Jenkinsfile:16]: What's the rationale for `<something>.plus(...).minus(<something>)`?

File: build.gradle [hunk 2]
```diff
@@ -38,7 +41,7 @@ plugins {
 }
 
 allprojects {
-    version "$confidential_id_release_version"
+    version "$confidential_id_release_version$confidential_id_versionSuffix"
```
Comment (yarivus) [inline build.gradle:44]: Shouldn't we have some delimeter between the version and the suffix, e.g. a dash or dot ?

File: constants.properties [hunk 3]
```diff
@@ -9,5 +9,10 @@ log4jVersion=2.17.1
 tokenReleaseGroup=com.r3.corda.lib.tokens
 tokenReleaseVersion=1.0
 confidentialIdReleaseGroup=com.r3.corda.lib.ci
-confidentialIdReleaseVersion=1.1
 artifactoryContextUrl=https://software.r3.com/artifactory
+confidentialIdReleaseVersion=1.1
+# This versionSuffix value should be left blank for an external release (GA) or set to -RC / -HC for internal releases
+# Also leave blank for standard release branch / feature branch activity
+versionSuffix=
+
```
Comment (yarivus) [inline constants.properties:17]: Remove redundant newlines

**Review by yarivus — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (ronanbrowne) [reply Jenkinsfile:16]: Copy and paste error thanks ...  Removed some redundant lines here. 

No need for Minus and also a duplicate 'build' entry in the list as this is already set via `nexusDefaultIqStage`

File: constants.properties (hunk 3 above)
Comment (ronanbrowne) [reply constants.properties]: updated

**Review by yarivus — APPROVED** [review]
LGTM

**Pull request merged**